File: Projects/Week 6 - Python/sentimental-mario-less/mario.py
# Uses get_int from CS50; if not in the CS50 environment, use input() and int() instead
from cs50 import get_int

# Prompt the user for the pyramid's height until a valid value is provided.
while True:
    height = get_int("Height: ")
    if height >= 1 and height <= 8:
        break

# Generate the pyramid
for i in range(1, height + 1):

    # Calculate how many spaces and hashes are needed for each row
    spaces = height - i
    hashes = i
    print(" " * spaces + "#" * hashes)

File: Projects/Week 1 - C/mario-less/mario.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    // Declare variable for height
    int height;

    // Prompt the user for the height until a valid number is given
    do
    {
        height = get_int("Height: ");
    }
    while (height < 1 || height > 8); // Ensure height is a positive integer

    // Loop to print the pyramid
    for (int i = 1; i <= height; i++) // Loop over each row
    {
        // Print leading spaces
        for (int j = 0; j < height - i; j++)
        {
            printf(" ");
        }

        // Print hashes
        for (int k = 0; k < i; k++)
        {
            printf("#");
        }

        // Move to the next line
        printf("\n");
    }
}
